### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by ArtiomTr**

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -3,146 +3,145 @@ import { context, getOctokit } from '@actions/github';
...
         } else {
-            throw new Error(
-                'Something went wrong! Looks like action runs in PR, but report for the base branch or pull_request information is missing!'
-            );
+            await generateCommitReport(reportContent!, context.repo, octokit);
+        }
+    });
+
+    await runStage('failedTestsAnnotations', dataCollector, async (skip) => {
+        if (!isInitialized || isHeadCoverageGenerated) {
```
Comment (sidharthv96) [inline src/index.ts:110]: isHeadCoverageGenerated will be true even if a test fails, so this step will be skipped.

Comment (sidharthv96) [reply src/index.ts:110]: https://github.com/sidharthv96/jest-coverage-report-action-test/pull/1/checks?check_run_id=3451046926#step:3:276

image

**Comment by sidharthv96** [comment]
image

This comes when a test run fails. 
I think we should handle that failure and not show "Unexpected error"

**Comment by ArtiomTr** [comment]
@sidharthv96 Thank you for the review! I will consider all your remarks.

**Comment by sidharthv96** [comment]
Hey, any areas I could help to get this pushed out faster?
Happy to pitch in :)

**Comment by ArtiomTr** [comment]
@sidharthv96, thank you for the initiative! As you can see, I have covered almost all of the files. There is only one file left untested - `run.ts`. It would be a huge help if you could write tests for this file. Thank you in advance.

**Comment by sidharthv96** [comment]
Added some tests in https://github.com/ArtiomTr/jest-coverage-report-action/pull/151

**Comment by ArtiomTr** [comment]
@sidharthv96 thanks a lot! Your tests look good, only one small change is required.

**Review by sidharthv96 — CHANGES_REQUESTED** [review]
coverageFile and baseCoverageFile options are not being utilized presently.

**Comment by ArtiomTr** [comment]
@sidharthv96, I think I will fix that in separate PR

**Pull request merged**